website_product_subscription/models/subscription_request.py

```diff
@@ -25,11 +27,11 @@ def send_gift_emails(self):
             ".gift_subscription_existing_user_mail_template"
         )
         for request in self:
-            if self.env["res.users"].user_exists(request.subscriber.email):
-                existing_user_template.send_mail(request.id)
+            if request.subscriber._has_web_access():
+                template = existing_user_template
             else:
-                new_user_template.send_mail(request.id)
-                request.subscriber.create_web_access()
+                template = new_user_template
+            template.send_mail(request.id)
             request.gift_sent = True
 
     @api.multi
```

Comment:
Could you refactor this function to `def create_subscriber_web_access` ?

I'm sorry, I could have sait it in my first review but I got confused twice :/

Reply:
+1